modelstore/storage/aws.py

```diff
@@ -88,28 +89,32 @@ def validate(self) -> bool:
     def _push(self, source: str, destination: str) -> str:
         logger.info("Uploading to: %s...", destination)
         self.client.upload_file(source, self.bucket_name, destination)
-        logger.debug("Finished: %s", destination)
         return destination
 
     def _pull(self, source: str, destination: str) -> str:
-        logger.info("Downloading from: %s...", source)
-        file_name = os.path.split(source)[1]
-        destination = os.path.join(destination, file_name)
-        self.client.download_file(self.bucket_name, source, destination)
-        logger.debug("Finished: %s", destination)
-        return destination
+        try:
+            logger.debug("Downloading from: %s...", source)
+            file_name = os.path.split(source)[1]
+            destination = os.path.join(destination, file_name)
+            self.client.download_file(self.bucket_name, source, destination)
+            return destination
+        except ClientError as e:
+            if int(e.response["Error"]["Code"]) == 404:
+                raise FilePullFailedException(e)
+            raise e
```

Comment:
There are different exceptions that can be raised if a file does not exist in one of the four types of storage (file system, aws s3, gcloud, azure). I've wrapped each individual one in a `FilePullFailedException` so that I can deal with model deletion in the parent class, without needing to know anything specific about the child class.